### PR TITLE
Refactored consumption.py

### PR DESCRIPTION
**Pull request by kaestro**
Haven't tested whether this runs for I haven't set up sqlite and don't know how to,

but I've started from refactoring consumption.py into separate files.

Wonder this type of programming won't be of your type of style.

If any suggestions and problems let me know.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,6 @@ env
 # notebooks
 .ipynb_checkpoints
 *.ipynb
+
+# venv
+venv
```
Comment (BuildWithData) [inline .gitignore:17]: pull request should be atomic, if you wanna update .gitignore do it, but in a separate pull request

File: src/db/create_tables.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    create_table_holdings_btc_bfill(logger, conn)
+    create_table_inflows_btc(logger, conn)
+    create_table_inflows_btc_bfill(logger, conn)
+    create_table_inflows_btc_bfill_inline(logger, conn)
+    create_table_inflows_btc_bxfill(logger, conn)
+    create_table_inflows_btc_sma5(logger, conn)
+
+
+def create_table_products(logger, conn):
+    PRODUCTS_CREATE = load_sql_file("db/queries/products_create.sql")
```
Comment (BuildWithData) [inline src/db/create_tables.py:18]: I like this idea of having all sql queries in a dedicated folder, but I would not hard code its path here, but rather define the path of the folder in the config file or as a constant and then each sql file path will simply be obtained by concatenating the folder path with the file name

**Comment by BuildWithData** [comment]
my plan is to do this eventually, but not right now bcs:

1. workflow works fine
2. there are other priorities: 
https://github.com/BuildWithData/BTC-ETF-Tracker/issues/20
https://github.com/BuildWithData/BTC-ETF-Tracker/issues/26
https://github.com/BuildWithData/BTC-ETF-Tracker/issues/21
https://github.com/BuildWithData/BTC-ETF-Tracker/issues/24
https://github.com/BuildWithData/BTC-ETF-Tracker/issues/14
https://github.com/BuildWithData/BTC-ETF-Tracker/issues/15

your refactor goes in the way I think we gonna do this eventually, but rather than functions I'd go with a ORM like in django. 

So we gonna have a class where we gonna define the interface and all the plumbing methods:

    class Table(ABC):
    
             def read():
                 """return all records from table"""
    
             def create():
                  """materialize empty table"""
    
             def write():
          
and then all tables will inherit from this and define the logic only, for example:

    class Products(Table):

         def __init__(self):
               self.schema = """
                    ticker      TEXT    NOT NULL    PRIMARY KEY,
                    type        TEXT    NOT NULL    CHECK (type in ('etf', 'etp')),
                    firm        TEXT    NOT NULL,
                    country     TEXT                CHECK (country in ('USA', 'CANADA'))
               """

so this won't be merged, but you can work on other tasks and keep this part of the repo like in this pull request if you like it. This is just a refactor, so you should get the same exact result I get when tables are created and populated.

Also added a few comments in your code

**Comment by BuildWithData** [comment]
> Haven't tested whether this runs for I haven't set up sqlite and don't know how to,
> 

sqlite db does not need any setting, that's why I have chose it. That said you can change its default settings, added [here](v) a comment from where you can start working this out 

Again, if you struggle with this, just let me know what is blocking you... on the other hand you can work on another issue, if this one is boring or too hard to fix

let me know :)

**Comment by kaestro** [comment]
Thanks for the comments. I didn't know that sqlite doesn't need any setups at all. That was why I was wondering where I can find the db from and reason I was refactoring the code because I'm not that good at reading code

I think I wasn't that blocked on the issue yet, but if it's in a hurry that it's really fine if you finish the work on this and I will work on another issue.

I won't try to refactor on those SQL embedded codes from now on, thanks for letting me know that you're planning on shifting towards orm

**Comment by kaestro** [comment]
@BuildWithData 

BTW, for its my first time working on open source project I'm not used to how I can be of communicating yet.

I thought sending pull request was only way of showing you what my idea was. Is there any better way of expressing my situation? it would be of real help. Thank you

**Comment by BuildWithData** [comment]
> I thought sending pull request was only way of showing you what my idea was. Is there any better way of expressing my situation? it would be of real help. Thank you

Look I am no expert at all, so don't worry, just ask whenever your are in doubt :)

coming to your question, I think it depends on the project itself, some people like it one way, others like it another way... it could also depend on project size, for example if many devs are working on a project, I think think having a pull request always linked to an issue is a good practice...

That said, I like the idea of draft pull request, you can learn more [here](https://github.blog/2019-02-14-introducing-draft-pull-requests/)

**Comment by BuildWithData** [comment]
> Thanks for the comments. I didn't know that sqlite doesn't need any setups at all. That was why I was wondering where I can find the db from and reason I was refactoring the code because I'm not that good at reading code

I think your was a smart move, reading code is one thing, writing/refactoring it just gives you a better understanding

> I think I wasn't that blocked on the issue yet, but if it's in a hurry that it's really fine if you finish the work on this and I will work on another issue.

yeah, if you could solve the issue on decimal digits, that would be great, but fell free to pick any other issue... I'd suggest to start with this https://github.com/BuildWithData/BTC-ETF-Tracker/issues/14, this is high priority and by working on it you would get a better understanding of the whole codebase

**Pull request closed without merging**